sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,118 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import platform
+import sys
+import importlib
+
+from .._build_utils import get_blas_info
+
+
+def _get_sys_info():
+    """System information
+
+    Return
+    ------
+    sys_info : dict
+        system and Python version information
+
+    """
+    python = sys.version.replace('\n', ' ')
+
+    blob = [
+        ("python", python),
+        ('executable', sys.executable),
+        ("machine", platform.platform()),
+    ]
+
+    return dict(blob)
+
+
+def _get_deps_info():
+    """Overview of the installed version of main dependencies
+
+    Returns
+    -------
+    deps_info: dict
+        version information on relevant Python libraries
+
+    """
+    deps = [
+        # (MODULE_NAME, f(mod) -> mod version)
+        ("pip", lambda mod: mod.__version__),
+        ("setuptools", lambda mod: mod.__version__),
+        ("sklearn", lambda mod: mod.__version__),
+        ("numpy", lambda mod: mod.version.version),
```

Comment:
These also support `mod.__version__` which seems somewhat universal.

Reply:
Not in older numpy versions, as far as I remember. Though maybe those versions are old enough not to be supported in scikit-learn anymore...

Reply:
I just checked, you're right @jnothman, `numpy.__version__` works in 1.8.2.